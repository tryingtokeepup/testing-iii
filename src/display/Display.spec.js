// Test away!
import React from 'React';
import { render, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';
import 'react-testing-library/cleanup-after-each';
import renderer from 'react-test-renderer';

import Display from './Display';

describe('<Display />', () => {
  it('matches', () => {
    //react
    const node = renderer.create(<Display />);
    expect(node.toJSON()).toMatchSnapshot();
  });

  // how about changing the snapshot that has different props passed in

  it('matches a snapshot with props passed into it', () => {
    const node = renderer.create(<Display locked={true} />);

    expect(node.toJSON()).toMatchSnapshot();
  });

  it('should use red-led class when locked', () => {
    const { getByText } = render(<Display locked={true} />);

    const lockedDisplay = getByText(/locked/i);

    expect(lockedDisplay).toHaveClass('red-led');
  });
  it('should use green-led class when unlocked', () => {
    const { getByText } = render(<Display locked={false} />);

    const lockedDisplay = getByText(/unlocked/i);

    expect(lockedDisplay).toHaveClass('green-led');
  });
});
