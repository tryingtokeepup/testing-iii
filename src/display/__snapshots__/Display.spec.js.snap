// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> matches 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;

exports[`<Display /> matches a snapshot with props passed into it 1`] = `
<div
  className="display panel"
>
  <div
    className="led red-led"
  >
    Locked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
